### PR TITLE
Prevent race condition for Store write operations

### PR DESCRIPTION
**Pull request by viren-nadkarni**
### Summary

Adds re-entrant lock for write operations to Stores to prevent races.

### Related

Should resolve https://github.com/localstack/localstack/issues/6682

### CODE REVIEW

**Review by giograno — APPROVED** [review]
LGTM 👍 Thanks for quicky jumping on this :)

File: localstack/services/stores.py [hunk 1]
```diff
@@ -210,18 +214,22 @@ def __getitem__(self, account_id: str) -> RegionBundle:
...
             raise ValueError(f"'{account_id}' is not a valid AWS account ID")
 
-        if account_id not in self.keys():
-            self[account_id] = RegionBundle(
-                service_name=self.service_name,
-                store=self.store,
-                account_id=account_id,
-                validate=self.validate,
-            )
+        with BUNDLE_LOCK:
```
Comment (thrau) [inline localstack/services/stores.py:217]: from my understand this linearizes *all* store access calls on one lock. meaning we can only ever make one concurrent access regardless of whether the bundles are related or not.

would it maybe make sense to add a lock for each bundle instead?

**Review by thrau — COMMENTED** [review]
hey! just stumbled across this and have a comment regarding concurrency/performance. we should be mindful of introducing global locks for things that happen per-rquest.

File: localstack/services/stores.py (hunk 1 above)
Comment (viren-nadkarni) [reply localstack/services/stores.py]: Great point @thrau, fixed in 77002c4574743cbe608b74faf45bfd8f6650a873

**Pull request merged**